ESQL: make order in TOP optional

mouhc1ne opened the pull request:
- Make optional the 3rd param to the TOP agg function, and default it to "asc" if omitted
- Add a capability
- Add a csv test
- Randomly drop the order field during unit test cases where the order is ascending
- Documentation changes

mouhc1ne committed:
ESQL: make order in TOP optional
- Make optional the 3rd param to the TOP agg function, and default it to "asc" if omitted
- Add a capability
- Add a csv test
- Randomly drop the order field during unit test cases where the order is ascending
- Documentation changes

elasticsearchmachine commented:
Pinging @elastic/es-analytical-engine (Team:Analytics)

dnhatn reviewed (commented):
I've left one comment, but the direction looks good. Thanks @mouhc1ne!

mouhc1ne committed:
ESQL: make order in TOP optional
- Make optional the 3rd param to the TOP agg function, and default it to "asc" if omitted
- Add a capability
- Add a csv test
- Randomly drop the order field during unit test cases where the order is ascending
- Documentation changes

dnhatn reviewed (approved):
LGTM, thanks @mouhc1ne!

ivancea reviewed (approved):
LGTM!
